### DISCUSSION TITLE
Want to generate submodule's schema in original directory node_modules.

### DISCUSSION THREAD

**Discussion by soumi-akira** [MichalLytek/typegraphql-prisma#171] [Q&A] [answered]
Let's say there is 3 repos like this

1. database (main database)
2. user-service (GraphQL endpoint)
3. product-service (GraphQL endpoint)

`schame.prisma` on database repo is like this.

```prisma
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "darwin", "debian-openssl-1.1.x"]
}

generator typegraphql {
  provider = "typegraphql-prisma"
}

# Definition of User and Product....
# ......
```

and database repo is added as submodule in other 2 repos.
Open user-service then run yarn `prisma generate --schema ./database/schema.prisma`, result is

```
✔ Generated Prisma Client (2.30.0) to .\node_modules\@prisma\client in 159ms

✔ Generated TypeGraphQL integration to .\database\node_modules\@generated\type-graphql
```

I think it is better to output generated code in root's `node_modules` like Prisma Client because it is odd to access the submodule's `node_modules`...

Is there some idea to solve this? Is it difficult to achieve behavior like Prisma Client? 

I know if I wrote `schame.prisma` like this, I can access the generated file even user or product service, but it is smarter if we can access the generated file like Prisma Client.

```prisma
# ......
generator typegraphql {
  provider = "typegraphql-prisma"
  output = "../generated/type-graphql"
}
# ......
```

**Comment by MichalLytek** [comment, marked as answer]
Just write the path manually to emit files in parent (root) `node_modules` 😉

**Reply by soumi-akira** [reply]
So you mean make `schame.prisma` like bellow right?

```prisma
# ......
generator typegraphql {
  provider = "typegraphql-prisma"
  output = "../node_modules/@generated/type-graphql"
}
# ......
```

Yes it might work, but in this case generated file is output outside of repository if you run `prisma generate` in `database` repo🤔

But, OK, I made some workaround like bellow. 

- copy.js
```js
const fs = require("fs-extra");
fs.copy(
  "./database/node_modules/@generated/type-graphql",
  "./node_modules/@generated/type-graphql",
  (err) => {
    if (err) return console.error(err);
    console.log("type files coiped.");
  }
);
```

- packege.json

```json
{
    "scripts": {
        "generate": "yarn prisma generate --schema ./hedgehog-video-database/prisma/schema.prisma && node copy-types.js"
    }
}
```
